src/z3c/baseregistry/browser/base.py

```diff
@@ -27,32 +33,30 @@
 BASENAME = _('-- Global Base Registry --')
 PARENTNAME = _('-- Parent Local Registry --')
 
-class BaseComponentsVocabulary(zope.schema.vocabulary.SimpleVocabulary):
-    """A vocabulary for ``IComponents`` utilities."""
 
-    zope.interface.classProvides(zope.schema.interfaces.IVocabularyFactory)
+@zope.interface.provider(zope.schema.interfaces.IVocabularyFactory)
+class BaseComponentsVocabulary(SimpleVocabulary):
+    """A vocabulary for ``IComponents`` utilities."""
 
     def __init__(self, context):
         terms = []
         utils = set()
-        
+
         # add available registry utilities
-        for name, util in \
-            zope.component.getUtilitiesFor(
-                zope.component.interfaces.IComponents, context):
-            
-            terms.append(zope.schema.vocabulary.SimpleTerm(util, name))
+        for name, util in component.getUtilitiesFor(IComponents, context):
+            terms.append(SimpleTerm(util, name))
             utils.add(util)
 
         # add location parent registry if any
-        lsm = [removeSecurityProxy(sm) for sm in context.__bases__ \
+        lsm = [removeSecurityProxy(sm)
+               for sm in context.__bases__
                if sm not in utils and ILocalSiteManager.providedBy(sm)]
-        if lsm:
-            terms.append(zope.schema.vocabulary.SimpleTerm(lsm[0], PARENTNAME))
+        lsm = [SimpleTerm(x, PARENTNAME) for x in lsm[:1]]
```

Comment:
It does make the logic a tiny bit harder to understand, so it gets a -0 from me.

I sometimes cheat this way too, so it doesn't get a -1.   ;)

If this makes 100% achievable and enables us to turn on the fail-if-coverage-drops-below-100% flag in whichever config file that is (tox.ini? .coveragerc?) to prevent test coverage regressions, it gets a deserved +1.